spec/services/bulk_invoice_service_spec.rb

```diff
@@ -47,4 +47,11 @@
       expect(filepath).to eq 'tmp/invoices/1234567.pdf'
     end
   end
+
+  describe "#orders_from" do
+    it "orders with completed desc" do
+      expect(service.send(:orders_from, [1, 2]).to_sql)
```

Comment:
This test doesn't execute the query and doesn't verify that it works. It also depends on the internal implementation. It's better to test objects at the interface where they interact with other objects, e.g. the renderer. I would to the following:

* Create three orders with varying `completed_at` dates, unsorted, e.g. 5 hours ago, 3 hours ago, 4 hours ago.
* Inject an InvoiceRenderer mock object.
* Verify that the renderer's `render_to_string` method is called with the newest order first and the oldest order last.

Reply:
Yes, it was a light test, no extra loads required but coverage was less, sure. I will check out your suggestions.